platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/storage/FileDirsPathsTask.kt

```diff
@@ -0,0 +1,25 @@
+package org.maplibre.android.storage
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class FileDirsPathsTask(
+	private val onCompletion: (resourcesCachePath: String, internalCachePath: String) -> Unit
+) {
+	fun execute(context: Context) {
+		CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Hmm, this reads a bit odd. I have to look up what it does... this is a global coroutine scope, right?

I.e. it runs app-wide, independent of any e.g. view lifecycle. Is this what you want to do? Because, like other Tasks, this one has a callback. So, the callback is fired no matter what, even if the listener should have been removed already (out of lifecycle).

(As for the other tasks, the Kotlin-idiomatic way would be to have this as a suspend function, i.e.
```kotlin
suspend fun getResourcesCachePath(context: Context): String = 
    withContext(IO) { FileSource.getCachePath(context) }
```
(etc)

Reply:
I agree.  If we weren't trying to support SDK 21, the suspend function approach would be much more elegant.